huaweicloud/services/cse/data_source_huaweicloud_cse_microservice_engines.go

```diff
@@ -0,0 +1,279 @@
+		},
+	}
+}
+
+func flattenMicroserviceEngineConfigCenterAddresses(engineDetail interface{}) []map[string]interface{} {
+	return []map[string]interface{}{
+		{
+			"private": utils.PathSearch("serviceEndpoint.kie.masterEntrypoint", engineDetail, nil),
+			"public":  utils.PathSearch("publicServiceEndpoint.kie.masterEntrypoint", engineDetail, nil),
+		},
+	}
+}
+
+func convertStrNumberToIntIgnoreErr(strNum string) int {
+	result, err := strconv.Atoi(strNum)
+	if err != nil {
+		log.Printf("[ERROR] unable to convert object from type string to type int: %s", err)
+		return 0
+	}
+	return result
+}
+
+func flattenAllMicroserviceEngines(engines []interface{}) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(engines))
+
+	for _, engineDetail := range engines {
+		result = append(result, map[string]interface{}{
+			"id":                         utils.PathSearch("id", engineDetail, nil),
+			"name":                       utils.PathSearch("name", engineDetail, nil),
+			"flavor":                     utils.PathSearch("flavor", engineDetail, nil),
+			"availability_zones":         utils.PathSearch("reference.azList", engineDetail, nil),
+			"network_id":                 utils.PathSearch("reference.networkId", engineDetail, nil),
+			"auth_type":                  utils.PathSearch("authType", engineDetail, nil),
+			"version":                    utils.PathSearch("version", engineDetail, nil),
+			"enterprise_project_id":      utils.PathSearch("enterpriseProjectId", engineDetail, nil),
+			"description":                utils.PathSearch("description", engineDetail, nil),
+			"eip_id":                     utils.PathSearch("reference.publicIpId", engineDetail, nil),
+			"extend_params":              utils.PathSearch("reference.inputs", engineDetail, nil),
+			"service_limit":              convertStrNumberToIntIgnoreErr(utils.PathSearch("reference.serviceLimit", engineDetail, "").(string)),
+			"instance_limit":             convertStrNumberToIntIgnoreErr(utils.PathSearch("reference.instanceLimit", engineDetail, "").(string)),
+			"service_registry_addresses": flattenMicroserviceEngineServiceRegistryAddresses(engineDetail),
+			"config_center_addresses":    flattenMicroserviceEngineConfigCenterAddresses(engineDetail),
+			"status":                     utils.PathSearch("status", engineDetail, nil),
+			"created_at": utils.FormatTimeStampRFC3339(int64(utils.PathSearch("createTime",
+				engineDetail, float64(0)).(float64))/1000, false),
+		})
+	}
+
+	return result
+}
+
+func dataSourceMicroserviceEnginesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.NewServiceClient("cse", region)
+	if err != nil {
+		return diag.Errorf("error creating CSE client: %s", err)
+	}
+
+	engines, err := queryMicroserviceEngines(client)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "CSE microservice engines")
```

Comment:
It is unnecessary to add `checkDeleted` in data source.

Reply:
done.